src/modules/flow/iio/nodes.c

```diff
@@ -184,9 +182,9 @@ gyroscope_close(struct sol_flow_node *node, void *data)
 {
     struct iio_direction_vector_data *mdata = data;
 
-    free((char *)mdata->config.trigger_name);
-    if (mdata->device)
-        sol_iio_close(mdata->device);
+    free((char *)mdata->iio_base.config.trigger_name);
+    if (mdata->iio_base.device)
+        sol_iio_close(mdata->iio_base.device);
```

Comment:
If you put `iio_base` as the _first_ element of each structure, as I showed you, you won't need a different `close` or `tick` for each node type. You can use:
```
struct iio_device_config *mdata = data;

free((char *)mdata->config.trigger_name);
if (mdata->device)
    sol_iio_close(mdata->device);
```
In a `common_close` method. Then, on JSON, you use only this `common_close` for all node types.